lib/src/platform/src/realtime/connection.dart

```diff
@@ -37,7 +37,8 @@ class Connection extends PlatformObject {
 
   /// A unique public identifier for this connection, used to identify this
   /// member.
-  String? id;
+  Future<String?> get id async =>
```

Comment:
It works, but this way we introduce breaking changes, and it's not align with the spec. Is there any chance we can subscribe to connection object changes (something similar to what we do for state)? I believe at least `id` and `errorReason` can't be changed without changing connection state, so we can simply add those fields in our listener

Reply:
We can do this, but there would be inconsistency for other similar fields. Not every field will change for change in the state. Change can be made subject to discussion -> https://ably-real-time.slack.com/archives/C030C5YLY/p1708586146096219?thread_ts=1708354649.094169&cid=C030C5YLY

Reply:
Okay, updated -> https://github.com/ably/ably-flutter/pull/503/commits/7609df0cda0e530b320e97a10d2f2bc2f7d2776e. This is still subject to how we decide to set class fields in the future.

Reply:
Please note that this can cause inconsistency in retrieved `id` value if accessed before setting it inside a callback. Imagine a scenario where we have 
realtime.connection.on('connected', ()=> {
id = realtime.connection.id; // this can return old `id` value since new one is yet to be updated via callback : (
}